dbt/include/trino/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,80 @@
+
+    {{ log("No existing view found, creating materialized view...", info=true) }}
+    {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+
+{% else %}
+    {%- if existing_relation.is_view %}
+        {%-set existing_type = 'view'%}
+    {%- elif existing_relation.is_table %}
+        {%-set existing_type = 'table'%}
+    {%- elif existing_relation.is_cte %}
+        {%-set existing_type = 'CTE'%}
+    {% endif %}
+
+    {%- if existing_type is not none and config_drop_any_existing == 'False' %}
+        {{ log("Found a " ~ existing_type ~ " with same name.", info=true) }}
+        {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+
+    {% elif existing_type is not none and config_drop_any_existing == 'True' and existing_type != 'table'%} -- theres an issue with starburst thinking that the materialised view is a table in the information schema, hence the logic on this line
+        {%- set drop_existing_sql = "DROP " ~ existing_type ~ " IF EXISTS " ~ existing_relation %}
+        {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+        {%- call statement('drop_existing') -%}
+            {{ drop_existing_sql }} 
+        {% endcall %}
+
+    {% else %}
+        {{ log("Found something with same name, trying to create materialized view anyway...", info=true) }}
+        {%- set build_sql = build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) %}
+    {% endif %}
+    
+{% endif %}
+
+-- Sends the sql command to starburst
+{%- call statement('main') -%}
+    {{ build_sql }} 
+{% endcall %}
+
+{{ run_hooks(pre_hooks) }}
+
+{{ return({'relations': [target_relation]}) }}
+
+{% endmaterialization %}
+
+---------------- Macros -------------------
+
+{%- macro build_materialized_view(target_relation,config_grace_period,config_max_import_duration,config_cron,config_refresh_interval) -%}
+{% if config_cron is not none and config_refresh_interval is not none %}
+    {{ log("Found config for CRON and Refresh Interval, error will be thrown as only 1 is allowed", info=true) }}
+    {{ exceptions.raise_compiler_error("Invalid config: only CRON or refresh interval allowed - " ~ config_cron ~ " - " ~ config_refresh_interval) }}
+
+{% else %}  
+    {% if config_cron is not none and config_refresh_interval is none %}
+        {%- set schedule_to_use ="cron = \'" ~  config_cron  ~ "\'," %}
+    {% else %}
+        {%- set schedule_to_use ="refresh_interval = \'" ~  config_refresh_interval  ~ "\'," %}
+    {% endif %}
+
+    {%- set sqlcode= "CREATE OR REPLACE MATERIALIZED VIEW " ~  target_relation ~ " WITH (
+    " ~ schedule_to_use ~ 
+    "grace_period = \'" ~  config_grace_period  ~ "\',
```

Comment:
Why the single quotes are escaped? I would think it is not required.

Reply:
Its so they land correct in starburst, we the escape it looks like this
WITH(
config_item = '1234',
)

without the single quotes it'll produce something like 
WITH(
config_item=1234,
)

Ive chose this format due to it working consistently and noticed issues with different values when the code is run manually on starburst, for example a value of 5.33h, so thought its safer keeping the '' in every query sent

Reply:
My point is within a double quoted string, I don't think it's needed to escape single quotes. We should definitely include the single quotes based on the type of the property.